notify-debouncer-full/src/lib.rs

```diff
@@ -455,7 +503,12 @@ impl<T: FileIdCache> DebounceDataInner<T> {
         }
 
         if let Some(target_queue) = self.queues.get_mut(&event.paths[0]) {
-            if !target_queue.was_created() {
+            if target_queue.was_removed() {
+                let event = target_queue.events.pop_front().unwrap();
```

Comment:
Yes, unwrap is safe here, I added a comment to explain.
Maybe we can refactor the code to make it more elegant and remove the unwrap, but the function doesn't return a Result, because it can only fail if there's an implementation error.